src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -53,12 +53,28 @@ def destroy
 
   private
 
-  def request_numbers
-    xml_hash.elements('number')
+  def set_request_numbers
+    @request_numbers = @parsed_xml.elements('number')
+    return if @request_numbers.present?
+
+    render_error(
+      status: 400,
+      errorcode: 'invalid_request',
+      message: 'Error while parsing the numbers of the requests'
+    )
   end
 
-  def xml_hash
-    Xmlhash.parse(request.body.read) || {}
+  def set_xml_hash
+    request_body = request.body.read
+    @parsed_xml = Xmlhash.parse(request_body) if request_body.present?
```

Comment:
I think we could make it a global validator used by all `update` actions on the API, what do you think @eduardoj ?

Reply:
I think the same. Everything that requires an xml should use validation of empty body and xml validation.

Reply:
We already have that: [ApplicationController.validate_xml_request](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/application_controller.rb#L380)

Reply:
Before I meant that everything that requires an xml should use validation of empty body and well-formed xml.

Method `validate_xml_request` validates the given text against a xml schema definition file (.xsd files). Maybe validation through xml schemas is too much at this point.

Reply:
It is indeed. All of this is still in flux.